kornia/augmentation/container/augmentation_sequential.py

```diff
@@ -0,0 +1,268 @@
+        self,
+        input: torch.Tensor,
+        item: nn.Module,
+        param: Optional[Dict[str, torch.Tensor]] = None,
+        itype: Union[str, int, DataCategory] = DataCategory.INPUT,
+    ) -> torch.Tensor:
+        if itype in [DataCategory.INPUT, DataCategory.MASK]:
+            return self.inverse_input(input, item, param)
+        if itype in [DataCategory.BBOX, DataCategory.BBOX_XYXY]:
+            return self.inverse_bbox(input, item, param, mode='xyxy')
+        if itype in [DataCategory.BBOX_XYHW]:
+            return self.inverse_bbox(input, item, param, mode='xyhw')
+        if itype in [DataCategory.KEYPOINTS]:
+            return self.inverse_keypoints(input, item, param)
+        raise NotImplementedError(f"input type of {itype} is not implemented.")
+
+    def inverse(
+        self,
+        *args: torch.Tensor,
+        params: Optional[Dict[str, Dict[str, torch.Tensor]]] = None,
+        input_types: Optional[List[Union[str, int, DataCategory]]] = None,
+    ) -> Union[torch.Tensor, List[torch.Tensor]]:
+        """Reverse the transformation applied.
+
+        Number of input tensors must align with the number of``input_types``. If ``input_types``
+        is not set, use ``self.input_types`` by default.
+        """
+        if input_types is None:
+            input_types = cast(List[Union[str, int, DataCategory]], self.input_types)
+        assert len(args) == len(input_types), (
+            "The number of inputs must align with the number of input_types, "
+            f"Got {len(args)} and {len(input_types)}."
+        )
+        self._params = {}
+        params = params if params is not None else {}
+
+        outputs = []
+        for input, itype in zip(args, input_types):
+            for item in list(self.children())[::-1]:
+                if isinstance(item, _AugmentationBase):
+                    func_name = item.__class__.__name__
+                    # Check if a param recorded
+                    param = self._params[func_name] if func_name in self._params else None
+                    # Check if a param provided. If provided, it will overwrite the recorded ones.
+                    param = params[func_name] if func_name in params else param
+                else:
+                    param = None
+                if isinstance(item, GeometricAugmentationBase2D) and itype in DataCategory:
+                    # Waiting for #1013 to specify the geometric and intensity augmentations.
+                    input = self.inverse_by_input_type(input, item, param, itype)
+                elif isinstance(item, IntensityAugmentationBase2D) and itype in DataCategory:
+                    pass  # Do nothing
+                else:
+                    raise NotImplementedError(f"input_type {itype} is not implemented for {item}.")
+            outputs.append(input)
+
+        if len(outputs) == 1:
+            return outputs[0]
+
+        return outputs
+
+    def forward(  # type: ignore
```

Comment:
Why type ignore?

Reply:
The function signature differs from the forward in ```nn.Module```. it complains.